components/brave_wallet/browser/cardano/cardano_knapsack_solver.h

```diff
@@ -0,0 +1,51 @@
+/* Copyright (c) 2025 The Brave Authors. All rights reserved.
+ * This Source Code Form is subject to the terms of the Mozilla Public
+ * License, v. 2.0. If a copy of the MPL was not distributed with this file,
+ * You can obtain one at https://mozilla.org/MPL/2.0/. */
+
+#ifndef BRAVE_COMPONENTS_BRAVE_WALLET_BROWSER_CARDANO_CARDANO_KNAPSACK_SOLVER_H_
+#define BRAVE_COMPONENTS_BRAVE_WALLET_BROWSER_CARDANO_CARDANO_KNAPSACK_SOLVER_H_
+
+#include <string>
+#include <vector>
+
+#include "base/types/expected.h"
+#include "brave/components/brave_wallet/browser/cardano/cardano_transaction.h"
+
+namespace brave_wallet {
+
+// Knapsack coin selection algorithm. Inspired by
+// https://github.com/sparrowwallet/drongo/blob/master/src/main/java/com/sparrowwallet/drongo/wallet/KnapsackUtxoSelector.java#L8
+// and
+// https://github.com/bitcoin/bitcoin/blob/v25.1/src/wallet/coinselection.cpp#L255
+// Tries to find the best set of inputs(minimal fee) for a transaction.
+// Does two runs of search: with and without change output. See
+// `SolveForTransaction` for details.
+// TODO(apaymyshev): consider moving this calculation to separate thread.
+class CardanoKnapsackSolver {
```

Comment:
I would say we can improve the comments for this class.

Reply:
Don't know actually what to add there. Seems good enough

Reply:
You can start from here

https://google.github.io/styleguide/cppguide.html#Class_Comments_Details
https://google.github.io/styleguide/cppguide.html#Function_Comments
https://google.github.io/styleguide/cppguide.html#Variable_Comments

The variable link has a section called *Class Data Members*.

Feel free to reach out in DMs if any help is needed.